graphdatascience/query_runner/neo4j_query_runner.py

```diff
@@ -81,14 +81,35 @@ def run_query(
 
             return df
 
-    def run_query_with_logging(
+    def call_endpoint(
+        self,
+        type: EndpointType,
+        endpoint: str,
+        yields: Optional[List[str]] = None,
+        body: Optional[str] = None,
+        params: Optional[Dict[str, Any]] = None,
+        database: Optional[str] = None,
+        custom_error: bool = True,
+    ) -> DataFrame:
+        call_keyword = "CALL" if type == EndpointType.PROCEDURE else "RETURN"
+
+        if yields is not None and type == EndpointType.FUNCTION:
+            raise ValueError("Functions cannot yield results")
```

Comment:
You'll need to specify what you mean by "function" here. Also, a client user may not know what a Cypher function is. If you don't put yields in the API yet, I think this should be a generic `RuntimeError`